plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/GameStateMachine.hs

```diff
@@ -165,6 +182,28 @@ scriptInstance = Scripts.Validator @(SM.StateMachine GameState GameInput)
 machineInstance :: SM.StateMachineInstance GameState GameInput
 machineInstance = SM.StateMachineInstance machine scriptInstance
 
+-- | Allocate the funds for each transition.
+allocate :: GameState -> GameInput -> Value -> ValueAllocation
+allocate (Initialised _) (ForgeToken _) currentVal = 
+    ValueAllocation
+        { vaOwnAddress    = currentVal
+        -- use 'Tx.forgeValue' to ensure that the transaction forges
+        -- the token.
+        , vaOtherPayments = Tx.forgeValue gameTokenVal
```

Comment:
Do we need a constraint that the token actually leaves the contract?

Reply:
If the token isn't explicitly allocated to the contract (in `currentVal`) then it automatically leaves the contract thanks to the balancing performed by the wallet.

When balancing, the newly minted token is counted on the left-hand side (as an input) and so it will be neutralised on the right-hand side by a public key output.

Reply:
Yes, I guess it will DTRT currently!